sqlair.go

```diff
@@ -365,5 +456,28 @@ func (tx *TX) Query(ctx context.Context, s *Statement, inputArgs ...any) *Query
 	}
 
 	qe, err := s.pe.Query(inputArgs...)
-	return &Query{qs: tx.tx, qe: qe, ctx: ctx, err: err}
+	if err != nil {
+		return &Query{ctx: ctx, err: err}
+	}
+
+	tx.db.prepedCacheMutex.Lock()
+	defer tx.db.prepedCacheMutex.Unlock()
+
+	var ok bool
+	var ps *sql.Stmt
+	ps, ok = tx.db.prepedCache.lookup(s, tx)
+	if !ok {
+		ps, ok = tx.db.prepedCache.lookup(s, nil)
+		if !ok {
```

Comment:
True. I've changed the order, though it does need to be added to the cache for both the branches so I left the `add` outside the branches.

Reply:
Ah right, the add also handles the position in the least recently used list. To me, `add` sounds like the element isn't already present in the cache, but it currently has two responsibilities:

- Adding new elements
- Bumping already-included elements to the start of the LRU list

Perhaps it would be cleaner to separate these out into two functions:
- `add` only adds new elements
- a new function, perhaps called `bump`, which moves the given entry to the front of the LRU list and is only used for existing elements

Reply:
I would argue that the current implementation makes for a pretty clean and simple interface for the cache, i.e. if you add an element to the cache you can expect it to be at the front of the queue, irrespective of whether it was in the cache before. By adding bump we would end up with conditionals and a few more lines of code every time we wanted to add something to the cache, because when add something we also always want to bump it.